.github/workflows/ci.yaml

```diff
@@ -0,0 +1,176 @@
+# This workflow will install Python dependencies, run tests and lint with a variety of Python versions
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-python-with-github-actions
+
+name: CI
+
+on:
+  push:
+    branches: [ $default-branch, add_github_actions ]
+  pull_request:
+    branches: [ $default-branch ]
+
+env:
+  DEFAULT_PYTHON: 3.7
+
+jobs:
+  lint-flake8:
+    name: Check flake8
+    runs-on: ubuntu-latest
+    steps:
+    - name: Check out code from Github
+      uses: actions/checkout@v2
+
+    - name: Set up Python ${{ env.DEFAULT_PYTHON }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ env.DEFAULT_PYTHON }}
+
+    - name: Cache dependencies
+      uses: actions/cache@v2
+      with:
+        path: ~/.cache/pip
+        key: ${{ runner.os }}-${{ env.DEFAULT_PYTHON }}-pip-flake8
+
+    - name: Install dependencies
+      run: |
+        pip install -U pip
+        pip install flake8
+
+    - name: Lint with flake8
+      run: |
+        # stop the build if there are Python syntax errors or undefined names
+        flake8 plexapi --count --select=E9,F63,F7,F82 --show-source --statistics
+        # The GitHub editor is 127 chars wide
+        flake8 plexapi --count --max-complexity=12 --max-line-length=127 --statistics
+
+
+  pytest:
+    name: pytest ${{ matrix.python-version }} (${{ matrix.plex }})
+    needs: lint-flake8
+    runs-on: ubuntu-latest
+    env:
+      PLEXAPI_AUTH_SERVER_BASEURL: http://127.0.0.1:32400
+      PLEX_CONTAINER_TAG: latest
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: [3.6, 3.7, 3.8, 3.9]
+        plex: ['unclaimed', 'claimed']
+        is-master:
+          - ${{ github.ref == 'refs/heads/master' }}
+        exclude:
+          - is-master: false
+            plex: claimed
+    steps:
+    - name: Check out code from Github
+      uses: actions/checkout@v2
+
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+
+    - name: Cache dependencies
+      uses: actions/cache@v2
+      with:
+        path: ~/.cache/pip
+        key: ${{ runner.os }}-${{ matrix.python-version }}-pip-${{ hashFiles('requirements_dev.txt') }}
+        restore-keys: ${{ runner.os }}-${{ matrix.python-version }}-pip-
+
+    - name: Install dependencies
+      run: |
+        pip install -U pip
+        pip install -r requirements_dev.txt
+        pip install -e .
+
```

Comment:
Add steps to cache the PMS docker image.
```yml
    - name: Get PMS Docker image digest
      id: docker-digest
      run: |
        mkdir -p ~/.cache/docker/${{ env.PLEX_CONTAINER }}
        echo "Image: ${{ env.PLEX_CONTAINER }}"
        echo "Tag: ${{ env.PLEX_CONTAINER_TAG }}"
        token=$(curl \
          --silent \
          "https://auth.docker.io/token?scope=repository:${{ env.PLEX_CONTAINER }}:pull&service=registry.docker.io" \
          | jq -r '.token')
        digest=$(curl \
          --silent \
          --header "Accept: application/vnd.docker.distribution.manifest.v2+json" \
          --header "Authorization: Bearer $token" \
          "https://registry-1.docker.io/v2/${{ env.PLEX_CONTAINER }}/manifests/${{ env.PLEX_CONTAINER_TAG }}" \
          | jq -r '.config.digest')
        echo "Digest: $digest"
        echo ::set-output name=digest::$digest

    - name: Cache PMS Docker image
      id: docker-cache
      uses: actions/cache@v2
      with:
        path: ~/.cache/docker/plexinc
        key: ${{ runner.os }}-docker-pms-${{ steps.docker-digest.outputs.digest }}

    - name: Pull PMS Docker image
      if: steps.docker-cache.outputs.cache-hit != 'true'
      run: |
        docker pull ${{ env.PLEX_CONTAINER }}:${{ env.PLEX_CONTAINER_TAG }}
        docker save -o ~/.cache/docker/${{ env.PLEX_CONTAINER }}-${{ env.PLEX_CONTAINER_TAG }}.tar ${{ env.PLEX_CONTAINER }}:${{ env.PLEX_CONTAINER_TAG }}
        echo "Saved image: ${{ env.PLEX_CONTAINER }}:${{ env.PLEX_CONTAINER_TAG }}"

    - name: Load PMS Docker image
      if: steps.docker-cache.outputs.cache-hit == 'true'
      run: |
        docker load -i ~/.cache/docker/${{ env.PLEX_CONTAINER }}-${{ env.PLEX_CONTAINER_TAG }}.tar
```

Reply:
Awesome!